HPXMLtoOpenStudio/resources/hpxml.rb

```diff
@@ -2944,6 +2946,35 @@ def delete
 
     def check_for_errors
       errors = []
+
+      if not @weekday_heating_setpoints.nil?
+        vals = @weekday_heating_setpoints.split(', ').map { |i| i.to_f }
```

Comment:
Good point, you are right; we don't need this code. When I remove this code, and just try running `hvac-base-programmable-thermostat-detailed.xml` with `WeekdaySetpointTempsHeatingSeason` set to length of 23 instead of 24, I get in the run.log: `Error: A 12-element array with 24-element arrays of numbers must be entered for the schedule.`